### PR TITLE
Fix #670: skip-backfill staleness bound, reason vocabulary polish

### LINKED ISSUE #670: [FOLLOW-UP] #657 hardening: skip-backfill staleness bound, reason vocabulary polish
Deferred findings from the #657 review pipeline (below the fix-now bar; grouped so none goes untracked):

1. **No staleness bound on the skip backfill.** `log-trade` skip rows inherit the LATEST candidate row regardless of age; `prune_candidates` (72h age-out) has no operational caller, so a skip logged days after the last scan silently inherits stale price/prob. Consider a scanned_at cutoff mirroring caddie-master's 48h research-void rule, or wire prune_candidates into the pipeline.
2. **`--reason` accepted on non-skip actions** — validates vocabulary only; an open/close row can carry a skip reason. Decide intent (reject vs allow) and pin with a test.
3. **`review_reject` reused for an infrastructure failure** (caddie-master.md decision-note-write-failure path) — a proceed that died to tooling, not review. Consider a distinct `infra_failed` value if the Pro agent's proceed-conversion query needs the distinction.
4. **Two caddie-master prose skip sites carry no `--reason`** (cooldown criterion 3 "price unchanged", criterion 4 "already traded") — land with reason ''; `cooldown` fits both. `price_moved`/`liquidity` are currently dead vocabulary (reserved).
5. **Scout skips with `--prob 0` and a real price** are retained in the missed-opportunity audit denominator with edge 0 (unknowable outcome) — pre-existing; assess whether prob-less skips should be excluded from the false-negative denominator entirely.

Context: #657 (fix adds candidate-row backfill for skips, `reason` column/migration v18, template rewrites, advisor degenerate-exclusion with auditable count).

### PR DESCRIPTION
**Pull request by jonnymobs907**
## Summary

Resolves the five #670 hardening items deferred from #657:

**1. Staleness bound on the skip backfill.** `log-trade` inherited skip analytics from the latest candidate row regardless of age. It now rejects rows scanned more than 48 hours ago — mirroring caddie-master's research-void rule — using the canonical `parse_scanned_at` parser (review finding: an inline `fromisoformat` had a latent timezone-corruption path for offset-bearing strings). Stale rows log at info; unparseable timestamps warn (data-integrity signal for Groundskeeper) and degrade to honest zeros. `prune_candidates` deliberately stays unwired: old candidate rows are still legitimately consumed by the cooldown display and the flip/STALE-CLOSE checks, and the reader-side bound makes them harmless for the backfill.

**2/3. Vocabulary: `infra_failed` and `already_traded`.** A decision-note write failure was logged as `review_reject` — a proceed that died to tooling, not a review verdict — and now uses `infra_failed`. "Already traded" skips (criterion 4) get their own value. Both are non-entry reasons: they keep honest zeros (the open trade row carries the real analytics) and stay out of the missed-opportunity audit. The non-entry set is now a single source of truth (`advisor.NON_ENTRY_SKIP_REASONS`) aliased by the CLI, with an identity drift-guard test — three review agents independently flagged the comment-only sync as armed-but-not-fired.

**4. Prose skip sites.** caddie-master criterion 3 (price unchanged) now explicitly instructs `--reason cooldown` via the criterion-2 template; criterion 4 gets a full `already_traded` log-trade template (no analytics flags, per the zero-flag pin).

**5. Audit denominator.** `analyze_missed_opportunities` now excludes skips missing **either** probability or price: log-trade's edge normalization zeroes edge unless both are positive, so such rows can never classify as missed wins and only dilute the false-negative rate — and legacy prob-only rows carried a fabricated constructor edge (probability minus a never-recorded price) that inflated the numerator. Both exclusions land in the auditable `excluded_degenerate` count.

`--reason` on non-skip actions (#670 item 2) was verified already rejected and pinned — no change.

Closes #670

## Test plan

- `test_cli_skip_analytics.py`: 3 staleness tests (>48h stale, 47h backfills — pins the comparison direction and the constant to ±1h, unparseable warns via caplog), non-entry loop derived from the shared constant.
- `test_advisor.py`: both half-degenerate shapes excluded with the fabricated-edge case pinned (`false_negative_rate` unchanged by legacy prob-only rows), price-only dilution mirror, non-entry loop derived from the constant, single-source-of-truth identity test.
- `test_agent_prompts.py`: fragments pin the `already_traded` template and the `infra_failed` instruction.
- Full suite: **1839 passed** on frozen code.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01XraN34AP4re87cAzt9LRKB

### CODE REVIEW

File: src/gimmes/cli.py [hunk 1]
```diff
@@ -2447,6 +2457,39 @@ async def _log() -> None:
...
+                    elif datetime.now(UTC) - scanned > timedelta(
+                        hours=_BACKFILL_MAX_AGE_HOURS,
+                    ):
+                        logging.getLogger(__name__).info(
+                            "candidate for %s scanned >%dh ago —"
+                            " too stale to backfill; analytics"
+                            " default to 0 (#670)",
+                            ticker, _BACKFILL_MAX_AGE_HOURS,
+                        )
+                        rows = []
```
[the remark this answers is not in the document]
Comment (jonnymobs907) [reply src/gimmes/cli.py:2495]: Fixed in f1e4a1b. Added a `cause_logged` flag: the stale, unparseable-timestamp, and lookup-error paths each log their own cause and suppress the fallback debug, which now fires only when the candidates table genuinely has no row for the ticker. The second review pass added DEBUG-level caplog regression tests for all three paths (including the previously untested sqlite3.Error branch).
